app/controllers/receive_package.js

```diff
@@ -133,11 +133,21 @@ export default Ember.Controller.extend({
       if(this.get("hasErrors")) {
         this.get("package").rollbackAttributes();
       }
-      this.transitionToRoute("review_offer.receive");
+      var loadingView = getOwner(this).lookup('component:loading').append();
+      var pkg = this.get("package");
+      pkg.set('inventoryNumber', null);
+      pkg.save()
```

Comment:
It doesn't seem right to me to fire an update request to remove inventory_number from package. 
How does the inventory number gets assigned in the first place??
It should only be assign on saving a package.

Also I don't see any delete request for generated inventory_number when we click cancel button.

Refer this => [Remove inventory number on cancel of creation of new item on Stock](https://github.com/crossroads/stock.goodcity/blob/47f654bd5f166ddeda395b3eac991b32bf2be8d7/app/controllers/items/new.js#L324)

Reply:
`print_barcode` method assigns `inventory_number` irrespective of whether you save it or not. That is why on canceling also it was showing `inventoryNumber`. Yes, I need to add delete `inventory_number`.